### PR TITLE
New Bosses, Updated Fire Ball, and Refactoring Huds.

### PR DESCRIPTION
**Pull request by Bailey-Ardrey**
Big Changes:

- 3 new bosses: Gohma, Dodongo, and Manhandla.
- Fire attack refinements
- InGameHud and InventoryHud are now singletons to make event handling easier from any class
- Add new Face Block enemies that take a fireball (I tweaked this so that they actually go into the Blocks list within Dungeon, so now they do not deal damage to Link... easier fix for me to implement quickly)

Small Changes:

- The first item Link picks up will automatically equip.
- Fix boomerang pause menu bug
- Fix bomb counting bug (bombs were not counting correctly)
- Pass the projectile to HitRegister() so that we can check what the projectile is prior to assigning damage (used for Gohma)
- Start power-up item implementation
- Remove the keyboard when the room is changed.

### CODE REVIEW

File: Sprintfinity3902/Entities/Link/Player.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class Player : AbstractEntity, IPlayer
         private static int THIRTEEN = 13;
         private static int ONE_HUNDRED_TWENTY = 120;
         private static int ONE_HUNDRED_NINETY_THREE = 193;
-        private static int INITIAL_HEALTH = 6;
+        private static int INITIAL_HEALTH = 10;
```
Comment (bradknez) [inline Sprintfinity3902/Entities/Link/Player.cs:23]: Eventually change this back, but fine for testing.

Comment (Bailey-Ardrey) [reply Sprintfinity3902/Entities/Link/Player.cs:23]: If we want to keep it at 6 for Master I am totally cool with that! I should have brought it up in the meeting. Kyle could keep it a 6 for his pr and that would be fine!

File: Sprintfinity3902/States/GameStates/WinState.cs [hunk 2]
```diff
@@ -16,18 +16,18 @@ public void Update(GameTime gameTime)
...
         public void Draw(SpriteBatch spriteBatch, GameTime gameTime)
         {
-            foreach (IHud hud in Game.huds)
-            {
-                hud.Draw(spriteBatch, Color.White);
-            }
+            Game.dungeonHud.Draw(spriteBatch, Color.White);
+            HudMenu.InGameHud.Instance.Draw(spriteBatch, Color.White);
+            HudMenu.InventoryHud.Instance.Draw(spriteBatch, Color.White);
+            Game.miniMapHud.Draw(spriteBatch, Color.White);
```
Comment (bradknez) [inline Sprintfinity3902/States/GameStates/WinState.cs:30]: This change makes a lot of sense to me! Nice!

**Review by bradknez — APPROVED** [review]
Good changes overall! Really like how the pickup works now and just the general changes to the HUD is nice. The bosses were very interesting to read through! Merging.

**Pull request merged**